test/unit/codegen/codegen_llvm_ir.cpp

```diff
@@ -474,6 +474,50 @@ SCENARIO("Unary expression", "[visitor][llvm]") {
     }
 }
 
+//=============================================================================
+// WhileStatement
+//=============================================================================
+
+SCENARIO("While", "[visitor][llvm]") {
+    GIVEN("Procedure with a simple while loop") {
+        std::string nmodl_text = R"(
+            FUNCTION loop() {
+                LOCAL i
+                i = 0
+                WHILE (i < 10) {
+                    i = i + 1
+                }
+                loop = 0
```

Comment:
Just a comment: in my mind I was expecting ‘loop = i’ so that this function can be called via jit and we check return value as 10.

Reply:
That's just IR check for correct branches/blocks. I do JIT testing in `codegen_llvm_execition.cpp`, and there return the result of the loop which I verify.